DDDEastAnglia.Tests/InMemoryRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace DDDEastAnglia.Tests
+{
+    internal class InMemoryRepository<T>
```

Comment:
Can `T` be constrained at all? At least to `class`, perhaps? An `InMemoryRepository<string>` might not be that valuable. (Or perhaps it might?)